app/controllers/HasCorrectGoodsController.scala

```diff
@@ -148,4 +163,72 @@ class HasCorrectGoodsController @Inject() (
             } yield Redirect(navigator.nextPage(HasCorrectGoodsCommodityCodeUpdatePage(recordId), mode, updatedAnswers))
         )
     }
+
+  private def processNewCommodityCode(mode: Mode, recordId: String, request: DataRequest[AnyContent], value: Boolean)(
+    implicit hc: HeaderCarrier
+  ) =
+    for {
+      // Save the user's answer
```

Comment:
I personally love comments, but should we have them? Up to you

Reply:
I'm hoping they are useful and demystify the code here, so I'm happy if no one is really unhappy 🙂

Reply:
i like the comments 😄